tests/integration/helpers.py

```diff
@@ -94,21 +95,38 @@ async def assert_resource_lxd_profile(unit: Unit, configs: dict[str, Any]) -> No
     assert disk == profile_content["devices"]["root"]["size"]
 
 
+async def get_runner_names(unit: Unit) -> tuple[str, ...]:
+    """Get names of the runners in LXD.
+
+    Args:
+        unit: Unit instance to check for the LXD profile.
+
+    Returns:
+        Tuple of runner names.
+    """
+    action = await unit.run("lxc list --format json")
+    await action.wait()
+
+    assert action.results["return-code"] == 0
```

Comment:
Just something to mind:
https://github.com/canonical/hardware-observer-operator/issues/36#issue-1795890406

```
# with juju 2
action = await unit.run(command, timeout=timeout)
results = action.results
return results # {'Code': '3', 'Stdout': 'inactive\n'}

# with juju 3
action = await unit.run(command, timeout=timeout)
await action.wait()
results = action.results
return results # {'return-code': 3, 'stdout': 'inactive\n'}

Reply:
Good catch.
The integration test currently uses juju 3.1.
Since it is likely we need to switch to juju 3.X in the future as juju 2.X goes out of support at some point, I think the current code is fine.